src/main/java/org/traccar/WebDataHandler.java

```diff
@@ -155,32 +181,86 @@ public String formatRequest(Position position) throws UnsupportedEncodingExcepti
     @Override
     protected Position handlePosition(Position position) {
 
-        String url;
-        if (json) {
-            url = this.url;
-        } else {
-            try {
-                url = formatRequest(position);
-            } catch (UnsupportedEncodingException | JsonProcessingException e) {
-                throw new RuntimeException("Forwarding formatting error", e);
+        class AsyncRequestAndCallback implements InvocationCallback<Response> {
+
+            private Integer delay = retryDelayMin;
+            private Map<String, Object> jsonPayload;
+            private Invocation.Builder requestBuilder;
+
+            AsyncRequestAndCallback(Position position) {
+
+                String formattedUrl;
+                try {
+                    formattedUrl = (json) ? url : formatRequest(position);
+                } catch (UnsupportedEncodingException | JsonProcessingException e) {
+                    throw new RuntimeException("Forwarding formatting error", e);
+                }
+
+                requestBuilder = client.target(formattedUrl).request();
+                if (header != null && !header.isEmpty()) {
+                    for (String line: header.split("\\r?\\n")) {
+                        String[] values = line.split(":", 2);
+                        requestBuilder.header(values[0].trim(), values[1].trim());
+                    }
+                }
+
+                if (json) {
+                    jsonPayload = prepareJsonPayload(position);
+                }
+
+                deliveryPendingCurrent.incrementAndGet();
+
+                send();
             }
-        }
 
-        Invocation.Builder requestBuilder = client.target(url).request();
+            private void send() {
+                if (json) {
+                    requestBuilder.async().post(Entity.json(jsonPayload), this);
+                } else {
+                    requestBuilder.async().get(this);
+                }
+            }
 
-        if (header != null && !header.isEmpty()) {
-            for (String line: header.split("\\r?\\n")) {
-                String[] values = line.split(":", 2);
-                requestBuilder.header(values[0].trim(), values[1].trim());
+            private void retry() {
+                try {
+                    deliveryFaliuresInRow.incrementAndGet();
+                    if (!retryEnabled || deliveryPendingCurrent.get() > deliveryPendingLimit) {
+                        deliveryPendingCurrent.decrementAndGet();
+                    } else {
+                        Integer i = 0;
+                        for ( ; i < delay; i++) {
+                            Thread.sleep(1000);
```

Comment:
I don't like what's going on here. Please use proper async scheduling, ideally with existing timer that we have.

Reply:
To be honest, now that I see it again, I don't like it either. Are you talking about using `GlobalTimer`? I hadn't seen it before. Thank you.

Reply:
Yes.